test/unit/cart.js

```diff
@@ -161,6 +191,28 @@ describe('Cart class', () => {
     });
   });
 
+  describe('sanitizeCheckout()', () => {
+    it('calls removeLineItems for all line items with deleted variants', () => {
```

Comment:
This is a bit of a nitpick, but I'm wondering if this and the test above are coupled too tightly to the implementation. Ideally, we just test behaviours. ex: We want to ensure that line items with null variants are removed, and you do do that, but checking the calls makes this very brittle if the implementation of how that's done ever changes. If the implementation changed, we'd still want to ensure the behaviour, but the test would still fail.

I'm curious what your thoughts are, and if you see a strong value in testing the implementation, if it'd be worthwhile to break that out into a separate test so the behavioural tests still work as insurance and are fully decoupled from internals.

Reply:
Since these tests require some stubbing, I think the implementation is always going to be somewhat implicitly tested based on what functions are being stubbed. I'm explicitly checking the calls for completeness, since we already have the stubs. If the implementation were to change they would most likely need to update the stubs anyways, even if the test was only checking for the behaviour.